### PR TITLE
DataAccessLayer

### PR DESCRIPTION
**Pull request by babanayak**
JUnit test for data access layer

### CODE REVIEW

File: src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		cutofftime1 = new CutOffTime();
+		cutofftime2 = new CutOffTime();
+		
+		cutofftime1.setCutoffTime("20:00");
+		cutofftime1.setProcessingDay("WEDNESDAY");
+
+		cutofftime2.setCutoffTime("20:00");
+		cutofftime2.setProcessingDay("THURSDAY");
+
+		List<CutOffTime> cptlist = new ArrayList<CutOffTime>();
```
Comment (vijamit) [inline src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java:89]: cptList -> cutOffTimeList

File: src/main/java/com/EstimatedDeliveryDate/Config/EstimatedDeliveryDateConfig.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package com.EstimatedDeliveryDate.Config;
+
```
Comment (vijamit) [inline src/main/java/com/EstimatedDeliveryDate/Config/EstimatedDeliveryDateConfig.java:2]: Generic comments : 
1. Interface methods are public/abstract by default, remove public
2. Use code formatter throughout code for proper formatting throughout the code.
https://stackoverflow.com/questions/15655126/how-to-auto-format-code-in-eclipse
3. Follow camel case naming convention through out the code, in test classes as well
4. Please following below for multine line comment
   /**
    *
    */
5. Don't include .class file in git pull request
6. Remove unnecessary lines from code
7. Single line comment should follow below pattern
    // Comment
8. cpt -> criticalPullTime

File: src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	@Test
+	public void testTransitTime() {
+		transittime = new TransitTime();
+		transittime.setShippingServiceId("ROYAL_MAIL_METHOD_1");
+		transittime.setTransitTime(840);
+		Map<String, Integer> map = new HashMap<String, Integer>();
+		map.put("US", 450);
+		map.put("IND", 360);
```
Comment (vijamit) [inline src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java:112]: Give better descriptive name to the variable.
Does map -> TransitTimeMap sound better?

File: src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+		holiday2.setProcessingHoliday(true);
+		holiday2.setHolidayDate("2020-05-23");
+
+		List<Holiday> holidaylist = new ArrayList<Holiday>();
+		holidaylist.add(holiday1);
+		holidaylist.add(holiday2);
+		holidays.setHolidayDatesList(holidaylist);
+		holidays.setShippingServiceId("ROYAL_MAIL_METHOD_1");
+		
+        when(mapper.load(Holidays.class,"ROYAL_MAIL_METHOD_1")).thenReturn(holidays);
```
Comment (vijamit) [inline src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java:73]: Indentation : Use code formatter

File: src/main/java/com/EstimatedDeliveryDate/Config/EstimatedDeliveryDateConfig.java (hunk 2 above)
Comment (babanayak) [reply src/main/java/com/EstimatedDeliveryDate/Config/EstimatedDeliveryDateConfig.java:2]: Made changes as you suggested and updated Amit

File: src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+import com.EstimatedDeliveryDate.Entity.CriticalPullTime;
+import com.EstimatedDeliveryDate.Entity.CutOffTime;
+import com.EstimatedDeliveryDate.Entity.Holiday;
+import com.EstimatedDeliveryDate.Entity.Holidays;
+import com.EstimatedDeliveryDate.Entity.TransitTime;
+import com.EstimatedDeliveryDate.Repository.EstimatedDeliveryDateRepositoryImpl;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+
+@ExtendWith(MockitoExtension.class)
+public class DataAccessTest {
```
[the remark this answers is not in the document]
Comment (babanayak) [reply src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java]: Changed it to EstimatedDeliveryDateRepositoryImplTest

File: src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java (hunk 1 above)
Comment (babanayak) [reply src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java]: Changed it to cutOffTimeList

File: src/main/java/com/EstimatedDeliveryDate/Entity/Holiday.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+@Setter
+@NoArgsConstructor
+@DynamoDBDocument
+public class Holiday implements Serializable {
+	private static final long serialVersionUID = 1L;
+	private String holidayDate;
+	private Boolean IsDeliveryHoliday;
+	private Boolean IsPickUpHoliday;
+	private Boolean IsProcessingHoliday;
```
Comment (vijamit) [inline src/main/java/com/EstimatedDeliveryDate/Entity/Holiday.java:21]: Follow camel case naming convention

File: src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java (hunk 4 above)
Comment (babanayak) [reply src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java]: Used codeformater and updated

File: src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java (hunk 3 above)
Comment (babanayak) [reply src/test/java/com/EstimatedDeliveryDate/Test/DataAccessTest.java]: Changed it to TransitTimeMap

File: src/test/java/com/EstimatedDeliveryDate/Test/EstimatedDeliveryDateReposiotryImplTest.java [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+		criticalPullTime.setProcessingDayCutoffList(cutOffTimeList);
+		criticalPullTime.setShippingServiceId("ROYAL_MAIL_METHOD_1");
+
+		when(mapper.load(CriticalPullTime.class, "ROYAL_MAIL_METHOD_1")).thenReturn(criticalPullTime);
+		assertEquals(repository.getCriticalPullTime("ROYAL_MAIL_METHOD_1"), criticalPullTime);
+
+	}
+
+	@Test
+	public void testgetTransitTime() {
```
Comment (vijamit) [inline src/test/java/com/EstimatedDeliveryDate/Test/EstimatedDeliveryDateReposiotryImplTest.java:101]: Try to use camel case in method name as well

Comment (babanayak) [reply src/test/java/com/EstimatedDeliveryDate/Test/EstimatedDeliveryDateReposiotryImplTest.java:101]: Ok

File: src/main/java/com/EstimatedDeliveryDate/Entity/Holiday.java (hunk 6 above)
Comment (babanayak) [reply src/main/java/com/EstimatedDeliveryDate/Entity/Holiday.java:21]: Ok

File: src/test/java/com/EstimatedDeliveryDate/Test/EstimatedDeliveryDateReposiotryImplTest.java (hunk 7 above)
Comment (babanayak) [reply src/test/java/com/EstimatedDeliveryDate/Test/EstimatedDeliveryDateReposiotryImplTest.java]: changed it and updated

**Pull request merged**